app/controllers/api/openid_connect/authorizations_controller.rb

```diff
@@ -0,0 +1,196 @@
+module Api
+  module OpenidConnect
+    class AuthorizationsController < ApplicationController
+      rescue_from Rack::OAuth2::Server::Authorize::BadRequest do |e|
+        logger.info e.backtrace[0, 10].join("\n")
+        render json: {error: e.message || :error, status: e.status}
+      end
+
+      before_action :authenticate_user!
+
+      def new
+        auth = Api::OpenidConnect::Authorization.find_by_client_id_and_user(params[:client_id], current_user)
+        if logged_in_before?(params[:max_age])
+          reauthenticate
+        elsif params[:prompt]
+          prompt = params[:prompt].split(" ")
+          handle_prompt(prompt, auth)
+        else
+          handle_authorization_form(auth)
+        end
+      end
+
+      def create
+        restore_request_parameters
+        process_authorization_consent(params[:approve])
+      end
+
+      def destroy
+        authorization = Api::OpenidConnect::Authorization.find_by(id: params[:id])
+        if authorization
+          authorization.destroy
+        else
+          raise ArgumentError, "Error while trying revoke non-existent authorization with ID #{params[:id]}"
+        end
+        redirect_to user_applications_url
+      end
+
+      private
+
+      def handle_prompt(prompt, auth)
+        if prompt.include? "select_account"
+          handle_prompt_params_error("account_selection_required",
+                                     "There is no support for choosing among multiple accounts")
+        elsif prompt.include? "none"
+          handle_prompt_none(prompt, auth)
+        elsif prompt.include?("login") && logged_in_before?(60)
+          reauthenticate
+        elsif prompt.include? "consent"
+          request_authorization_consent_form
+        else
+          handle_authorization_form(auth)
+        end
+      end
+
+      def reauthenticate
+        sign_out current_user
+        params_as_get_query = params.map {|key, value| key.to_s + "=" + value }.join("&")
+        authorization_path_with_query = new_api_openid_connect_authorization_path + "?" + params_as_get_query
```

Comment:
`new_api_openid_connect_authorization_path(params)` should produce the correct query string.